### DISCUSSION TITLE
Running files with %run

### DISCUSSION THREAD

**Discussion by psychemedia** [jupyterlite/jupyterlite#866] [Q&A]
I'm trying to run files from my own JupyterLite instances using magic console commands of the form `%run myfile.py`.

If I clone and build the `jupyterlite-demo` repo, then publish it via Github Pages using the provided Github Action, things work fine; but if I then modify the `requirements.txt` file, eg by adding the xeus-python kernel, and republish to the same Pages site via the Action, when I try to refresh the environment and run the console command, I get a file not found (only tested in Chrome)?

```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
File /lib/python3.10/site-packages/IPython/core/magics/execution.py:696, in ExecutionMagics.run(self, parameter_s, runner, file_finder)
    695     fpath = arg_lst[0]
--> 696     filename = file_finder(fpath)
    697 except IndexError as e:

File /lib/python3.10/site-packages/IPython/utils/path.py:90, in get_py_filename(name)
     89         return py_name
---> 90 raise IOError("File `%r` not found." % name)

OSError: File `'test.py'` not found.

The above exception was the direct cause of the following exception:

Exception                                 Traceback (most recent call last)
Cell In [1], line 1
----> 1 get_ipython().run_line_magic('run', 'test.py')

File /lib/python3.10/site-packages/IPython/core/interactiveshell.py:2364, in InteractiveShell.run_line_magic(self, magic_name, line, _stack_depth)
   2362     kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2363 with self.builtin_trap:
-> 2364     result = fn(*args, **kwargs)
   2365 return result

File /lib/python3.10/site-packages/IPython/core/magics/execution.py:707, in ExecutionMagics.run(self, parameter_s, runner, file_finder)
    705     if os.name == 'nt' and re.match(r"^'.*'$",fpath):
    706         warn('For Windows, use double quotes to wrap a filename: %run "mypath\\myfile.py"')
--> 707     raise Exception(msg) from e
    708 except TypeError:
    709     if fpath in sys.meta_path:

Exception: File `'test.py'` not found.
```

Checking the dev/tools network log, I get a call to `/api/drive` returning a *Status Code: 405  / Referrer Policy: strict-origin-when-cross-origin* error?

Even if I clear all the caches, saved storage etc, reload the environment, then recreate the test `.py` file,  I get the same error. So it seems to be that my republishing step is somehow breaking things in the file handling of the deployed environment???

Some insight into what I'm doing wrong, or what is likely happening, would be really useful...

**Comment by jtpio** [comment]
> If I clone and build the jupyterlite-demo repo, then publish it via Github Pages using the provided Github Action, things work fine

Do you mean the `%run` magic works fine the first time the site is deployed?

**Reply by psychemedia** [reply]
Yes.. then it seems hit or miss whether it continues to work if I reload. I can't reliably replicate the issue.
